### PR TITLE
DSD-984: useNYPLBreakpoints hook

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-984](https://jira.nypl.org/browse/DSD-984)

## This PR does the following:

- Creates a `useNYPLBreakpoints` hook to replace manually doing conditional checks with the `useWindowSize` hook and using magic numbers for breakpoint values.
- Now, instead of setting a setting a temporary value to "600", using `useEffect`, and the width from `useWindowSize`, this hook returns all of that combined into a single boolean value.

```
const {
  isLargerThanSmall,
  isLargerThanMedium,
  isLargerThanLarge,
  isLargerThanXLarge
} = useNYPLBreakpoints();
```

And now this can be used such as `{isLargerThanMedium ? <DesktopView /> : <MobileView />}`, for example.

## How has this been tested?

Locally in storybook and unit tests.
Note: a file was brought in from Chakra to help mocking the `MediaMatch` API.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/ButtonGroup/ButtonGroup.tsx [hunk 1]
```diff
@@ -39,25 +39,9 @@ export const ButtonGroup = chakra(
...
-        setFinalButtonWidth("full");
-        setFinalLayout("column");
-      } else {
-        // Otherwise, set the layout and button width to the values
-        // passed in via the `buttonWidth` and `layout` props.
-        setFinalButtonWidth(buttonWidth);
-        setFinalLayout(layout);
-      }
-    }, [buttonWidth, layout, windowDimensions.width]);
+    const { isLargerThanMedium } = useNYPLBreakpoints();
```
Comment (bigfishdesign13) [inline src/components/ButtonGroup/ButtonGroup.tsx]: `isLargerThanMedium` is used three times right here in this file.  And I am guessing that is predominantly what I am going to see throughout the rest of the code changes, as that is the breakpoint we target for mobile-to-desktop UI format changes.  Would it make sense to also include something like `isLargerThanMobile`?

File: src/hooks/useNYPLBreakpoints.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * This hook is used to determine if the current screen size is larger than
+ * the specific NYPL breakpoint values. The returned value is an object with
+ * boolean values for each breakpoint.
+ */
+const useNYPLBreakpoints = () => {
+  const [
+    isLargerThanSmall,
+    isLargerThanMedium,
```
Comment (bigfishdesign13) [inline src/hooks/useNYPLBreakpoints.ts:11]: Echoing my comment from `ButtonGroup.tsx`, would it make sense to also include something like isLargerThanMobile in addition to these four base options?

**Review by bigfishdesign13 — COMMENTED** [review]
I really like this.  I have often wondered what we could add to eliminate the hardcoded breakpoint number values in the code.  

Ultimately, this looks good, but I have a question about a possible addition.

**Comment by EdwinGuzman** [comment]
Updated with a `isLargerThanMobile` boolean value.

**Review by bigfishdesign13 — APPROVED** [review]
Awesome.  Ship it!

**Pull request merged**